### PR TITLE
Added verbose printing of vocab to confirm correct data loading

### PR DESCRIPTION
**Pull request by hobson**
- @NSriram5 started on dependency-free parameter and vocab serialization
- @NSriram5 added verbose printing of vocab for debugging
- @hobson deleted redundant parens from dataset_loader.rs
- @hobson did manual regression testing
  - Vocab seems identical
  - Model training converges to similar loss after 100 epochs
  - Chatbot can't answer simple questions from the training dataset (seems underfit)

### CODE REVIEW

**Comment by anshumanpatil** [comment]
@hobson  Thanks merged
